ext/native/thin3d/VulkanRenderManager.cpp

```diff
@@ -42,24 +42,39 @@ void CreateImage(VulkanContext *vulkan, VkCommandBuffer cmd, VKRImage &img, int
 	vkCreateImage(vulkan->GetDevice(), &ici, nullptr, &img.image);
 
 	VkMemoryRequirements memreq;
-	vkGetImageMemoryRequirements(vulkan->GetDevice(), img.image, &memreq);
+
+	bool dedicatedAllocation = false;
+	if (vulkan->DeviceExtensions().KHR_dedicated_allocation) {
+		VkImageMemoryRequirementsInfo2KHR memReqInfo2{VK_STRUCTURE_TYPE_IMAGE_MEMORY_REQUIREMENTS_INFO_2_KHR};
+		memReqInfo2.image = img.image;
+
+		VkMemoryRequirements2KHR memReq2 = {VK_STRUCTURE_TYPE_MEMORY_REQUIREMENTS_2_KHR};
+		VkMemoryDedicatedRequirementsKHR memDedicatedReq{VK_STRUCTURE_TYPE_MEMORY_DEDICATED_REQUIREMENTS_KHR};
+		memReq2.pNext = &memDedicatedReq;
+
+		vkGetImageMemoryRequirements2KHR(vulkan->GetDevice(), &memReqInfo2, &memReq2);
+
+		memreq = memReq2.memoryRequirements;
+		dedicatedAllocation =
+			(memDedicatedReq.requiresDedicatedAllocation != VK_FALSE) ||
+			(memDedicatedReq.prefersDedicatedAllocation != VK_FALSE);
+	} else {
+		vkGetImageMemoryRequirements(vulkan->GetDevice(), img.image, &memreq);
+	}
```

Comment:
This code duplicated three times really says "helper" to me...

-[Unknown]

Reply:
Yup, was just too lazy...  Will extract one at some point.